frontend/src/components/common/IconButton.tsx

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { Box, Button, IconButton as ChakraIconButton } from "@chakra-ui/react";
+
+interface IconButtonProps {
+  color: string;
+  hoverColor: string;
+  onClick: () => void;
+  icon: React.ReactElement;
+}
+
+const IconButton = ({
+  onClick,
+  color,
+  hoverColor,
+  icon,
+}: IconButtonProps): React.ReactElement => {
+  return (
+    <Box _hover={{ color: hoverColor }} color={color}>
```

Comment:
Also is there any reason we need to make an extra box here? I feel like you might be able to pass these directly on the button, but I'm missing context and can't test so I might be wrong